scripts/zones/Aht_Urhgan_Whitegate/npcs/Naja_Salaheem.lua

```diff
@@ -11,9 +11,14 @@ require("scripts/zones/Aht_Urhgan_Whitegate/TextIDs");
 require("scripts/globals/missions");
 require("scripts/globals/keyitems");
 require("scripts/globals/titles");
+require("scripts/globals/npc_util");
 -----------------------------------
 
 function onTrade(player,npc,trade)
+    if (npcUtil.tradeHas(trade, 2163) and player:getVar("Promotion") == 1) then -- Rank to PFC
+        player:confirmTrade()
```

Comment:
move this player:confirmTrade() into the elseif csid == 5002 block on line 193.  That way, the traded items are only taken once the cutscene finishes and the keyItem is obtained.